crates/runtime/src/dataconnector/spiceai.rs

```diff
@@ -211,6 +233,28 @@ impl DataConnector for SpiceAI {
         Some(read_write_result)
     }
 
+    fn supports_changes_stream(&self) -> bool {
+        true
+    }
+
+    fn changes_stream(&self, table_provider: Arc<dyn TableProvider>) -> Option<ChangesStream> {
+        let federated_table_provider_adaptor = table_provider
+            .as_any()
+            .downcast_ref::<FederatedTableProviderAdaptor>()?;
+        let flight_table = federated_table_provider_adaptor
+            .table_provider
+            .as_ref()?
+            .as_any()
+            .downcast_ref::<FlightTable>()?;
```

Comment:
We shouldn't pass in a table provider here and try to downcast it. We already have the flight client from the `FlightFactory` on `self.flight_factory`